app/src/main/kotlin/at/bitfire/davdroid/ui/DelayedSyncManager.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.ui
+
+import android.accounts.Account
+import at.bitfire.davdroid.sync.worker.SyncWorkerManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class DelayedSyncManager @Inject constructor(
+    private val syncWorkerManager: SyncWorkerManager
+) {
+
+    private val delayJobs: ConcurrentHashMap<Account, Job> = ConcurrentHashMap()
+    private val applicationScope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
```

Comment:
Could we avoid the [potentially expensive singleton](https://dagger.dev/dev-guide/#reusable-scope) by putting the properties into a companion object?

Reply:
I don't think so, as we won't gain anything from that. A companion object is always a singleton itself, so it doesn’t reduce memory allocation, if that's what you mean by it being expensive? I guess we could use `@Reusable` to give dagger more flexibility, but I am not sure we won't hit any issues when working with multiple accounts in the ConcurrentHashMap then.